controller/rest/delete.go

```diff
@@ -84,18 +85,29 @@ func (s *Server) DeleteVolume(rw http.ResponseWriter, req *http.Request) error {
 		replicas DeletedReplicas
 		wg       sync.WaitGroup
 	)
+	logrus.Info("Delete volume")
 	apiContext := api.GetApiContext(req)
+	id := mux.Vars(req)["id"]
+	v := s.getVolume(apiContext, id)
+	if v == nil {
+		rw.WriteHeader(http.StatusNotFound)
+		return fmt.Errorf("%v", volumeNotFound)
+	}
 	replicaCount := len(s.c.ListReplicas())
 	if replicaCount == 0 {
```

Comment:
What does it mean to have a volume with zero replica's?

Reply:
It's possible that volume structure is not updated which contains the various links related to the operations such as shutdown, start and rebuild. So to reverify the events checking the replica count. Should not we check replica count ?